### DISCUSSION TITLE
Show Time (HH:MM:SS) for datetime data types in Table Preview and DAX Query Results

### DISCUSSION THREAD

**Discussion by corycundy** [TabularEditor/TabularEditor3#1279] [Ideas]
### Is your feature request related to a problem? Please describe

The problem is that the time components are not displayed of datetime data types in the Table Preview, DAX Query results and possibly other areas of the application. While this might seem like an esthetic request, it actually would have saved me a great deal of time in troubleshooting a problem. I wrote a DAX measure that created a date based on parts using the DATE function and then using that date to query a fact table that had time components, which I didn't realize at the time. This was causing the measure to return unexpected results. I often rely on the data preview for quick checks of the data. If I would have seen time components in the data preview, the issue would have been immediately apparent. It was a quick fix once I realized it, but it would have made my realization much faster.  I realized that I missed a CAST to date in my view. Easy fix, once I found it.

### Describe the solution you'd like

I would like to see the full values of datetime columns in the Data Preview, DAX Query Results and any other similar areas of the application. If there is existing reasons for only showing the date part of the datetime, a setting in preferences to determine if the time component shows or not in the various parts of the application would be helpful.

### Goals

1. Better insight to the full value of datatime columns
2. Eliminate the hiding of potential existing time data.

### Background

The alternative that I use now is to view the time portion of a datetime is to take the time to open another tool, such DAX Studio and run the same query. SSMS even shows the time component. While this works, it isn't part of the natural development workflow. In my case above, it didn't help until I realized what the problem was. Then I opened DAX Studio to confirm.

**Comment by mlonsk** [comment]
Hi @corycundy 

I have done some testing and I think there are a few things going on here. 

1. We have a bug in the DAX Query where the seconds are not being shown as seen in this screenshot:
![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/3f9872c2-7ed6-429b-893d-39aa8237bd33)
I have created an issue to track [this](https://github.com/TabularEditor/TabularEditor3/issues/1283)

2. If there is not Format string on your DateTime column the Data Preview tries to figure out if the column data includes Time - here there seems to be an issue that this mechanism does not work correctly:
![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/19e7926a-a2cb-440a-ade8-8d2a82cde2a5)
I have created an issue to track [this](https://github.com/TabularEditor/TabularEditor3/issues/1282) 

However, if you set a Format string on your column (Which is considered a best practice in general) the DateTime will display correctly in the the Data Preview:
![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/8fbbc402-1589-44d6-82b6-b2d445289166)

The Pivot Grid works as expected with the DateTime columns and measures in my testing.

Are you experiencing other behavior than what I have described above?

**Reply by corycundy** [reply]
Morten, thanks for digging in!

These are exactly the issues I am experiencing. Thanks for adding both issues.

Cory Cundy

**Reply by mlonsk** [reply]
The issues where fixed in 3.17.0 but we forgot to close this discussion. Doing so now.
